include/seqan3/range/view/interleave.hpp

```diff
@@ -0,0 +1,352 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Joshua Kim <joshua.kim AT fu-berlin.de>
+ * \brief Provides seqan3::view::interleave.
+ */
+
+#pragma once
+
+#include <cmath>
+
+#include <range/v3/view/split.hpp>
+#include <range/v3/view/join.hpp>
+
+#include <seqan3/core/metafunction/pre.hpp>
+#include <seqan3/core/metafunction/transformation_trait_or.hpp>
+#include <seqan3/range/view/all.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+
+namespace seqan3::detail
+{
+
+// ============================================================================
+//  view_interleave
+// ============================================================================
+
+/*!\brief The type returned by seqan3::view::interleave.
+ * \tparam urng_t The type of the underlying range, must model std::ranges::RandomAccessRange.
+ * \implements std::ranges::View
+ * \implements std::ranges::RandomAccessRange
+ * \implements std::ranges::SizedRange
+ * \ingroup view
+ *
+ * \details
+ *
+ * Note that most members of this class are generated by ranges::view_interface which is not yet documented here.
+ */
+template <std::ranges::RandomAccessRange urng_t, std::ranges::RandomAccessRange inserted_rng_t>
+    //!\cond
+    requires std::ranges::View<urng_t> && std::ranges::SizedRange<urng_t> &&
+             std::ranges::View<inserted_rng_t> && std::ranges::SizedRange<inserted_rng_t> &&
+             std::Same<reference_t<urng_t>, reference_t<inserted_rng_t>>
+    //!\endcond
+class view_interleave : public ranges::view_interface<view_interleave<urng_t, inserted_rng_t>>
+{
+private:
+    //!\brief The underlying range.
+    urng_t urange;
+    //!\brief The step size for the insertion.
+    size_t step_size{};
+    //!\brief The range to be inserted into urange.
+    inserted_rng_t inserted_range;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief The reference_type.
+    using reference         = reference_t<urng_t>;
+    //!\brief The const_reference type is equal to the reference type.
+    using const_reference   = detail::transformation_trait_or_t<seqan3::reference<urng_t const>, void>;
+    //!\brief The value_type (which equals the reference_type with any references removed).
+    using value_type        = value_type_t<urng_t>;
+    //!\brief If the underliying range is Sized, this resolves to range_type::size_type, otherwise void.
```

Comment:
wrong description, underlying range is guaranteed to be sized as per the requirements